validator_test.go

```diff
@@ -14335,3 +14335,317 @@ func TestValidateFn(t *testing.T) {
 		Equal(t, fe.Tag(), "validateFn")
 	})
 }
+func TestMapStructBasicValidation(t *testing.T) {
+	type Inner struct {
+		Value string `validate:"max=5"`
+	}
+	type Outer struct {
+		Data map[string]Inner `validate:"dive"`
+	}
+	obj := Outer{
+		Data: map[string]Inner{
+			"key1": {Value: "exceeds"},
+		},
+	}
+	validate := New()
+	err := validate.Struct(obj)
+	if err == nil {
+		t.Fatal("Expected validation error due to Value > 5 chars")
+	}
```

Comment:
Let's replace all 'if err == nil => t.Fatal' in new tests and add comments to keep things consistent.
```suggestion
	NotEqual(t, err, nil)
```